capture/capture_otbn.py

```diff
@@ -0,0 +1,764 @@
+    """ Setup target, scope, and project.
+
+    Args:
+        cfg: The configuration for the current experiment.
+        project: The path for the project file.
+
+    Returns:
+        The target, scope, and project.
+    """
+    # Calculate pll_frequency of the target.
+    # target_freq = pll_frequency * target_clk_mult
+    # target_clk_mult is a hardcoded constant in the FPGA bitstream.
+    cfg["target"]["pll_frequency"] = cfg["target"]["target_freq"] / cfg[
+        "target"]["target_clk_mult"]
+
+    # Init target.
+    logger.info(f"Initializing target {cfg['target']['target_type']} ...")
+    target = CWFPGA(bitstream=cfg["target"]["fpga_bitstream"],
+                    force_programming=cfg["target"]["force_program_bitstream"],
+                    firmware=cfg["target"]["fw_bin"],
+                    pll_frequency=cfg["target"]["pll_frequency"],
+                    baudrate=cfg["target"]["baudrate"],
+                    output_len=cfg["target"]["output_len_bytes"],
+                    protocol=cfg["target"]["protocol"])
+
+    # Init scope.
+    scope_type = cfg["capture"]["scope_select"]
+
+    # Determine sampling rate, if necessary.
+    cfg[scope_type]["sampling_rate"] = determine_sampling_rate(cfg, scope_type)
+    # Convert number of cycles into number of samples, if necessary.
+    cfg[scope_type]["num_samples"] = convert_num_cycles(cfg, scope_type)
+    # Convert offset in cycles into offset in samples, if necessary.
+    cfg[scope_type]["offset_samples"] = convert_offset_cycles(cfg, scope_type)
+
+    logger.info(
+        f"Initializing scope {scope_type} with a sampling rate of {cfg[scope_type]['sampling_rate']}..."  # noqa: E501
+    )
+
+    # Create scope config & setup scope.
+    scope_cfg = ScopeConfig(
+        scope_type=scope_type,
+        acqu_channel=cfg[scope_type].get("channel"),
+        ip=cfg[scope_type].get("waverunner_ip"),
+        num_samples=cfg[scope_type]["num_samples"],
+        offset_samples=cfg[scope_type]["offset_samples"],
+        sampling_rate=cfg[scope_type].get("sampling_rate"),
+        num_segments=cfg[scope_type]["num_segments"],
+        sparsing=cfg[scope_type].get("sparsing"),
+        scope_gain=cfg[scope_type].get("scope_gain"),
+        pll_frequency=cfg["target"]["pll_frequency"],
+    )
+    scope = Scope(scope_cfg)
+
+    # OTBN's public-key operations might not fit into the sample buffer of the scope
+    # These two parameters allows users to conrol the sampling frequency
+    # `adc_mul` affects the sample frequency (clock_freq = adc_mul * pll_freq)
+    # `decimate` is the ADC downsampling factor that allows us to sample at
+    #  every `decimate` cycles.
+    if scope_type == "husky":
+        if "adc_mul" in cfg["husky"]:
+            scope.scope.scope.clock.adc_mul = cfg["husky"]["adc_mul"]
+        if "decimate" in cfg["husky"]:
+            scope.scope.scope.adc.decimate = cfg["husky"]["decimate"]
+        # Print final scope parameter
+        print(
```

Comment:
```suggestion
        logger.info(
```
I suggest to stick to `logger.info` inside the script.